bestiary/core/api.py

```diff
@@ -157,3 +211,161 @@ def delete_ecosystem(ctx, ecosystem_id):
     ecosystem.id = ecosystem_id
 
     return ecosystem
+
+
+@django.db.transaction.atomic
+def delete_project(ctx, project_id):
+    """"""
+    if project_id is None:
+        raise InvalidValueError(msg="'project_id' cannot be None")
+
+    trxl = TransactionsLog.open('delete_project', ctx)
+
+    try:
+        project = find_project(project_id)
+    except ValueError as e:
+        raise InvalidValueError(msg=str(e))
+    except NotFoundError as exc:
+        raise exc
+
+    delete_project_db(trxl, project=project)
+
+    trxl.close()
+
+    # Setting former ID manually, as it can't be referenced once it has been removed
+    project.id = project_id
+
+    return project
+
+
+@django.db.transaction.atomic
+def link_parent_project(ctx, project_id, parent_id):
+    """"""
+    if project_id is None:
+        raise InvalidValueError(msg="'project_id' cannot be None")
+    if parent_id is None:
+        raise InvalidValueError(msg="'parent_id' cannot be None")
+
+    trxl = TransactionsLog.open('link_projects', ctx)
+
+    try:
+        project = find_project(project_id)
+        parent_project = find_project(parent_id)
+    except ValueError as e:
+        raise InvalidValueError(msg=str(e))
+    except NotFoundError as exc:
+        raise exc
+
+    link_parent_project_db(trxl, project, parent_project)
+
+    trxl.close()
+
+    return project
+
+
+@django.db.transaction.atomic
+def unlink_parent_project(ctx, project_id):
+    """"""
+    if project_id is None:
+        raise InvalidValueError(msg="'project_id' cannot be None")
+
+    trxl = TransactionsLog.open('link_projects', ctx)
+
+    try:
+        project = find_project(project_id)
+    except ValueError as e:
+        raise InvalidValueError(msg=str(e))
+    except NotFoundError as exc:
+        raise exc
+
+    unlink_parent_project_db(trxl, project)
+
+    trxl.close()
+
+    return project
```

Comment:
It's better to define this as `move_project`. So you don't need to have two functions doing practically the same. Also you can add here the ecosystem to do the same at once. Having the next prototype for the function:

`def move_project(ctx, from_project_id, to_project_id=None, to_ecosystem_id=None)`

we can have the next behaviour:

- When `to_project_id` and `to_ecosystem_id` are `None`, the project will be a root project.
- When `to_project_id` is not None, the project will be a child of `to_project_id`
- When `to_ecosystem_id` is not None, the project will be moved to that ecosystem (as a root or not depending if `to_project_id` was given).

Reply:
Ok @sduenas, thank you for the idea!